common/persistence/cassandra/queue_v2_store.go

```diff
@@ -233,27 +244,82 @@ func (q *queueV2Store) CreateQueue(
 	return &persistence.InternalCreateQueueResponse{}, nil
 }
 
-func (q *queueV2Store) getMinMessageID(
+func (s *queueV2Store) RangeDeleteMessages(
+	ctx context.Context,
+	request *persistence.InternalRangeDeleteMessagesRequest,
+) (*persistence.InternalRangeDeleteMessagesResponse, error) {
+	if request.InclusiveMaxMessageMetadata.ID < persistence.FirstQueueMessageID {
+		return nil, fmt.Errorf(
+			"%w: id is %d but must be >= %d",
+			persistence.ErrInvalidQueueRangeDeleteMaxMessageID,
+			request.InclusiveMaxMessageMetadata.ID,
+			persistence.FirstQueueMessageID,
+		)
+	}
+	q, err := s.getQueue(ctx, request.QueueType, request.QueueName)
+	if err != nil {
+		return nil, err
+	}
+	minMessageID := s.getPartitionMinMessageID(q.Metadata, request.QueueType, request.QueueName)
+	maxID := request.InclusiveMaxMessageMetadata.ID
+	err = s.session.Query(
+		TemplateRangeDeleteMessagesQuery,
+		request.QueueType,
+		request.QueueName,
+		0, // partition
+		minMessageID,
+		maxID,
+	).WithContext(ctx).Exec()
+	if err != nil {
+		return nil, gocql.ConvertError("QueueV2RangeDeleteMessages", err)
+	}
+	partitions := q.Metadata.Partitions
+	if len(partitions) != 1 {
```

Comment:
I don't think this code is safe for multi-partition queues? The request may want to delete messages for a different partition, but the code will always delete from partition 0?

Reply:
Modified to just return an error here and elsewhere.